src/lib.rs

```diff
@@ -104,42 +105,64 @@ fn git_short_hash() -> Result<String, Box<dyn std::error::Error>> {
     Ok(git_short_hash)
 }
 
-/// Run git with given arguments, as if it was run from the workspace directory,
-/// and try to parse the resulting stdout with given function.
-/// Returns a formatted error with stdout or stderr on any error.
-fn run_git<T, E: Display>(
-    args: &[&str],
-    parse: impl Fn(&str) -> Result<T, E>,
-) -> Result<T, Box<dyn std::error::Error>> {
-    const WORKSPACE_DIR: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/../..");
-
-    let command = || format!("git -C {WORKSPACE_DIR} {args}", args = args.join(" "));
-    let output = Command::new("git").args(["-C", WORKSPACE_DIR]).args(args).output()?;
-
+fn extract_stdout<'a>(
+    cmd_line: &'a str,
```

Comment:
`cmd_line`의 수명(lifetime)이 `output`이나 반환값과 같을 필요는 없을 듯합니다:

```suggestion
    cmd_line: &'_ str,
```